sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationGatewayListenerImpl.java

```diff
@@ -237,7 +247,15 @@ public ApplicationGatewayListenerImpl withHttps() {
 
     @Override
     public ApplicationGatewayListenerImpl withHostname(String hostname) {
-        this.innerModel().withHostname(hostname);
+        this.innerModel().withHostname(null);
+        // future work: add hostname instead of replacing
+        if (hostname == null) {
+            this.innerModel().withHostNames(null);
```

Comment:
Are we allowed to have no hostname in listener?

Ideally if we want to clear a hostname, code should provide a `withoutHostname` in update flow.